cf-via-components/code/handler.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const Component = require('@serverless/aws-s3');
+const { Context, extractInputs, extractState, handlerWrapper } = require('./utils');
+
+function handler(event) {
+  const inputs = extractInputs(event);
+  // NOTE: apparently OldResourceProperties is only passed-in when
+  // dealing with Update requests --> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+  // therefore we re-use the inputs as 'state' when dealing with `Delete` requests
+  let state = {};
+  if (event.RequestType === 'Update') {
+    state = extractState(event);
```

Comment:
In my understanding what is extracted here is not really a _state_, but previous _input_ (one that was provided as _input_ to custom resource with previous successful stack deployment), so it's not a value in any way output by component.

When we're after a _state_, I think we need to, store, output from component, somewhere in S3 (or DynamoDb) storage,  and with every consecutive deployment retrieve it, pass it to component, and update it with value returned by it (at least that's how I think generic approach to components state handling could look).

But, maybe I misunderstood something?